### PR TITLE
HPCC-23398 Provide containerized HPCC log example

### PR DESCRIPTION
**Pull request by rpastrana**
- Provides containerized HPCC logging information via README
- Provides hpcc log processing via Elastic Helm information
- Provides hpcc log processing via Azure AKS Insights info

signed-off-by: Rodrigo Pastrana <rodrigo.pastrana@lexisnexisrisk.com>

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [x] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [ ] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [ ] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [ ] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [x] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

### CODE REVIEW

**Comment by rpastrana** [comment]
@jake @ming please review

**Comment by richardkchapman** [comment]
@jakesmith @xwang2713

File: helm/examples/logging/README.md [hunk 1]
```diff
@@ -0,0 +1,56 @@
+# Containerized HPCC Systems Logging
+
+## HPCC Systems logging background
+
+Containerized HPCC Systems components provide informative application level logs for the purpose of debugging problems, auditing actions, and progress monitoring. Following the most widely accepted containerized methodologies, those logs are written to the standard error (stderr) stream. At the node level, the contents of the standard error and out streams are redirected to a target location by a container engine. In a Kubernets environment, the Docker container engine redirects the streams to a logging driver, which Kubernetes configures to write to a file in JSON format. The logs are exposed by Kubernets via the aptly named "logs" command.
```
Comment (jakesmith) [inline helm/examples/logging/README.md:5]: couple of typo' Kubernets -> Kubernetes

File: helm/examples/logging/README.md [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+Many tools are available to help create an appropriate solution based on either a do-it-yourself approach, or managed features available from cloud providers.
+
+For the simplest of environments, it might be acceptable to rely on the standard Kubernetes process which forwards all contents of stdout/stderr to file. However, as the complexity of the cluster grows or the importance of retaining the logs' content grows, a cluster-level logging architecture should be employed.
+
+Cluster-level logging for the containerized HPCC Systems cluster can be accomplished by including a logging agent on each node. The task of each of agents is to expose the logs or push them to a log processing backend. Logging agents are generally not provided out of the box, but there are several available such as Elasticsearch and Stackdriver Logging. Various cloud providers offer built-in solutions which automatically harvest all stdout/err streams and provide dynamic storage and powerful analytic tools, and the avility to create custom alerts based on log data.
```
Comment (jakesmith) [inline helm/examples/logging/README.md:19]: >The task of each of agents is

The task of each of agent is

File: helm/examples/logging/README.md [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+
+By default, the component logs are not filtered, and contain the following columns:
+    
+    MessageID TargetAudience DateStamp TimeStamp ProcessId ThreadID QuotedLogMessage
+
+The logs can be filterd by TargetAudience, or by Category, and the output columns can be configured. Logging configuration settings can be applied at the global, or component level.
+
+### Target Audience Filtering
+
+The availble target audiences include operator(OPR), user(USR), programmer(PRO), audit(ADT), or all. The filter is controled by the `<section>`.logging.audiences value. The string value is comprised of 3 letter codes delimited by the aggregation operator (+) or the removal operator (-).
```
Comment (jakesmith) [inline helm/examples/logging/README.md:35]: is '\<section\>' supposed to be there?

File: helm/examples/logging/README.md [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+    
+    For example, all component log output to include Programer and User messages only
```
Comment (jakesmith) [inline helm/examples/logging/README.md:37]: Programmer

File: helm/examples/logging/azure/README.md [hunk 5]
```diff
@@ -0,0 +1,32 @@
+# HPCC Log Processing via Azure's AKS Insights
+
+Azure's AKS Insights is an optional feature designed to help monitor performance and health of kubernetes based clusters. 
+Once enabled and associated with a given AKS with an active HPCC System cluster, the HPCC component logs are automatically captured by Insights since all STDERR/STDOUT data is captured and made available for monitoring and/or querying purposes. As is usually the case with cloud provider features, cost is a significant consideration and should be well understood by user. Log content is written to the logs store associated with your Log Analytics workspace.
```
Comment (jakesmith) [inline helm/examples/logging/azure/README.md:4]: >automatically captured by Insights since all

automatically captured by Insights, since all

Comment (jakesmith) [reply helm/examples/logging/azure/README.md:4]: >be well understood by user.

by the user.

File: helm/examples/logging/azure/README.md [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+
+The AKS Insights interface on Azure provides Kubernets-centric cluster/node/container-level health metrics visualizations, and direct links to container logs via "log analytics" interfaces. The logs can be queried via “Kusto” query language (KQL).
```
Comment (jakesmith) [inline helm/examples/logging/azure/README.md:6]: > Kubernets

Kubernetes

File: helm/examples/logging/azure/README.md [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+    | where ContainerID in (ContainerIdList)
+    | project LogEntrySource, LogEntry, TimeGenerated, Computer, Image, Name, ContainerID
+    | order by TimeGenerated desc
+    | render table
+
+    Sample output
+    > 6/20/2020, 1:00:00.244 AM	stderr	1 TxSummary[activeReqs=6 auth=NA contLen=352 rcv=0ms handleHttp=3ms user=@10.240.0.4 req=POST wsstore.SET v1.0 total=3ms ] 	aks-default-12315622-vmss00000i			bc1555515a09e12a129c3ea5df0b76fb74c4227354dc2b643182c8f910b33ed4	
+	  > 6/20/2020, 12:59:58.910 AM	stderr	1 TxSummary[activeReqs=5 auth=NA contLen=99 rcv=0ms handleHttp=2ms user=@10.240.0.5 req=POST wsstore.FETCH v1.0 total=2ms ] 	aks-default-12315622-vmss00000i			bc1555515a09e12a129c3ea5df0b76fb74c4227354dc2b643182c8f910b33ed4
+
+More complexed queries can be formulated to fetch specific information provided in any of the log columns including unformatted data in the log message. The Insights interface fasilitates creation of alerts based on those queries, which can be used to trigger emails, SMS, Logic App execution, and many other actions.
```
Comment (jakesmith) [inline helm/examples/logging/azure/README.md:24]: > complexed

complex

> fasilitates

facilitates

**Comment by jakesmith** [comment]
@rpastrana - looks good/read well. Only spotted a few typo' type issues.

File: helm/examples/logging/README.md (hunk 2 above)
Comment (mckellyln) [reply helm/examples/logging/README.md:19]: avility

**Comment by rpastrana** [comment]
Thanks @jakesmith I believe all issues were addressed. Next time, I'll make sure to run the readme over a spell checker.

File: helm/examples/logging/README.md [hunk 8]
```diff
@@ -0,0 +1,57 @@
...
+
+## HPCC Systems application level logging details
+
+As mentioned earlier, the HPCC Systems logs provide a wealth of information which can be used for benchmarking, auditing, debugging, monitoring, etc. The type of information provided in the logs and its format is trivially controlled via standard Helm configuration.
+
+By default, the component logs are not filtered, and contain the following columns:
+    
+    MessageID TargetAudience DateStamp TimeStamp ProcessId ThreadID QuotedLogMessage
+
+The logs can be filterd by TargetAudience, or by Category, and the output columns can be configured. Logging configuration settings can be applied at the global, or component level.
```
Comment (jakesmith) [inline helm/examples/logging/README.md:31]: > filterd

filtered

**Comment by jakesmith** [comment]
@rpastrana - spotted 1 more typo'. 

Worth squashing ready for merge after that.

File: helm/examples/logging/README.md (hunk 8 above)
Comment (richardkchapman) [reply helm/examples/logging/README.md:31]: Also looks like rather a long line?

Comment (jakesmith) [reply helm/examples/logging/README.md:31]: Other lines are longer (e.g. line 19),
but they format ok (MD) when viewed in e.g. github, e.g.: https://github.com/hpcc-systems/HPCC-Platform/blob/2bdc2bba6defbc87c7d9611ad027522a9a2b86ef/helm/examples/logging/README.md

**Comment by rpastrana** [comment]
@richardkchapman please merge.

**Pull request merged**